pkg/scheduler/schedule_one.go

```diff
@@ -89,10 +90,43 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 	podsToActivate := framework.NewPodsToActivate()
 	state.Write(framework.PodsToActivateKey, podsToActivate)
 
-	schedulingCycleCtx, cancel := context.WithCancel(ctx)
-	defer cancel()
-	scheduleResult, assumedPodInfo := sched.schedulingCycle(schedulingCycleCtx, state, fwk, podInfo, podsToActivate, start)
-	if scheduleResult.FeasibleNodes == 0 {
+	var scheduleResult ScheduleResult
+	var assumedPodInfo *framework.QueuedPodInfo
+
+	func() {
```

Comment:
Why we need to wrap it in a function?

Reply:
Just for readability here. We can easy to tell the start and end of schedulingCycle here. Also cancel the context as early as possible. Also, the schedulingCycleCtx can only be used inside this block.

Reply:
Organize the error handling into a method for both.